copybookreader-plugins/src/main/java/co/cask/hydrator/plugin/batch/CopyBookSource.java

```diff
@@ -0,0 +1,283 @@
+    @Description("The schema for the data as it will be formatted in CDAP. Sample schema: " +
+      "{\n" +
+      "    \"type\": \"record\",\n" +
+      "    \"name\": \"schemaBody\",\n" +
+      "    \"fields\": [\n" +
+      "        {\n" +
+      "            \"name\": \"name\",\n" +
+      "            \"type\": \"string\"\n" +
+      "        },\n" +
+      "        {\n" +
+      "            \"name\": \"age\",\n" +
+      "            \"type\": \"int\"\n" +
+      "        }" +
+      "    ]\n" +
+      "}")
+    private String schema;
+
+    @Nullable
+    private Integer fileStructure;
+
+    @Name("copybookContents")
+    @Description("Contents of the COBOL copybook file which will contain the data structure. Eg: \n" +
+      "000100*                                                                         \n" +
+      "000200*   DTAR020 IS THE OUTPUT FROM DTAB020 FROM THE IML                       \n" +
+      "000300*   CENTRAL REPORTING SYSTEM                                              \n" +
+      "000400*                                                                         \n" +
+      "000500*   CREATED BY BRUCE ARTHUR  19/12/90                                     \n" +
+      "000600*                                                                         \n" +
+      "000700*   RECORD LENGTH IS 27.                                                  \n" +
+      "000800*                                                                         \n" +
+      "000900        03  DTAR020-KCODE-STORE-KEY.                                      \n" +
+      "001000            05 DTAR020-KEYCODE-NO      PIC X(08).                         \n" +
+      "001100            05 DTAR020-STORE-NO        PIC S9(03)   COMP-3.               \n" +
+      "001200        03  DTAR020-DATE               PIC S9(07)   COMP-3. ")
+    private String copybookContents;
+
+    public CopybookSourceConfig(String referenceName, String copybookContents, String binaryFilePath,
+                                @Nullable String schema, @Nullable Integer fileStructure, @Nullable Long maxSplitSize) {
+      super(referenceName);
+      this.copybookContents = copybookContents;
+      this.binaryFilePath = binaryFilePath;
+      this.schema = schema;
+      this.fileStructure = fileStructure == null ? CopybookConstants.DEFAULT_FILE_STRUCTURE : fileStructure;
+      this.maxSplitSize = maxSplitSize == null ? CopybookConstants.DEFAULT_MAX_SPLIT_SIZE : maxSplitSize;
+    }
+
+    public CopybookSourceConfig() {
+      super(String.format("CopybookReader"));
+      fileStructure = CopybookConstants.DEFAULT_FILE_STRUCTURE;
+    }
+
+    @VisibleForTesting
+    public final int getFileStructure() {
+      return fileStructure;
+    }
+    /**
+     * Validate the configuration parameters required
+     */
+    private void validate() {
+      if (schema != null) {
+        parseSchema();
+      }
+      //check the file extension
+      if (binaryFilePath.length() < 4) {
```

Comment:
I mentioned last time that we shouldn't be enforcing this. It is not good to assume all files end with '.bin', as people can name files whatever they want.